testing/fil_token_integration/actors/test_actor/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+                    return_ipld(&receipt)
+                }
+                TestAction::Burn => {
+                    // burn the tokens
+                    let burn_params = BurnParams {
+                        amount: token_params.amount,
+                    };
+                    let receipt = sdk::send::send(&Address::new_id(sdk::message::caller()), method_hash!("Burn"), RawBytes::serialize(&burn_params).unwrap(), TokenAmount::zero()).unwrap();
+                    if !receipt.exit_code.is_success() {
+                        panic!("burn call failed");
+                    }
+                    NO_DATA_BLOCK_ID
+                }
+            }
+        },
+        "Action" => {
+            // take action independent of the receiver hook
+            let params: ActionParams = deserialize_params(input);
+
+            // get our balance
+            let get_balance = || {
+                let self_address = Address::new_id(sdk::message::receiver());
+                let balance_receipt = sdk::send::send(&params.token_address, method_hash!("BalanceOf"), RawBytes::serialize(self_address).unwrap(), TokenAmount::zero()).unwrap();
+                if !balance_receipt.exit_code.is_success() {
+                    panic!("unable to get balance");
+                }
+                balance_receipt.return_data.deserialize::<TokenAmount>().unwrap()
+            };
+
+            match params.action {
+                TestAction::Accept => {
+                    // nothing to do here
+                    NO_DATA_BLOCK_ID
+                }
+                TestAction::Reject => {
+                    // nothing to do here
+                    NO_DATA_BLOCK_ID
+                }
+                TestAction::Transfer(to, operator_data) => {
+                    // transfer to a target address
+                    let balance = get_balance();
+                    let transfer_params = TransferParams {
+                        to,
+                        amount: balance,
+                        operator_data,
+                    };
+                    let receipt = sdk::send::send(&params.token_address, method_hash!("Transfer"), RawBytes::serialize(&transfer_params).unwrap(), TokenAmount::zero()).unwrap();
+                    // ignore failures at this level and return the transfer call receipt so caller can decide what to do
+                    return_ipld(&receipt)
+                }
+                TestAction::Burn => {
+                    // burn the tokens
+                    let balance = get_balance();
+                    let burn_params = BurnParams {
+                        amount: balance,
+                    };
+                    let receipt = sdk::send::send(&params.token_address, method_hash!("Burn"), RawBytes::serialize(&burn_params).unwrap(), TokenAmount::zero()).unwrap();
+                    if !receipt.exit_code.is_success() {
```

Comment:
Same as the Receive hook version, explode here or fail quietly and let the caller figure it out?

Reply:
I reckon fail for both